cmake/macros/testWrapper.py

```diff
@@ -166,6 +184,56 @@ def _diff(fileName, baselineDir, verbose):
 
     return True
 
+def _imageDiff(fileName, baseLineDir, verbose, warn=None, warnpercent=None,
+               hardwarn=None, fail=None, failpercent=None, hardfail=None,
+               perceptual=None):
+    import platform
+    if platform.system() == 'Windows':
+        imageDiff = 'idiff.exe'
+    else:
+        imageDiff = 'idiff'
+
+    cmdArgs = []
+    if warn is not None:
+        cmdArgs.extend(['-warn', warn])
+
+    if warnpercent is not None:
+        cmdArgs.extend(['-warnpercent', warnpercent])
+
+    if hardwarn is not None:
+        cmdArgs.extend(['-hardwarn', hardwarn])
+
+    if fail is not None:
+        cmdArgs.extend(['-fail', fail])
+
+    if failpercent is not None:
+        cmdArgs.extend(['-failpercent', failpercent])
+
+    if hardfail is not None:
+        cmdArgs.extend(['-hardfail', hardfail])
+
+    if perceptual:
+        cmdArgs.extend(['-p'])
+
+    for image in glob.glob(fileName):
+        cmd = [imageDiff]
+        cmd.extend(cmdArgs)
+        cmd.extend([_resolvePath(baseLineDir, image), image])
```

Comment:
One problem I ran into was that this code assumes that idiff will be in the user's PATH. In my case, I ran tests on a Linux machine from a build done with build_usd.py and that did not pick up the idiff that was built for me, so I had to go manually update my PATH before running the tests. Typically this isn't needed -- the test registrations all supply enough info for the tests to run (although I'm not sure this is true on Windows).

At CMake time, we do get the full path to idiff through the FindOpenImageIO.cmake module -- it should be set under the CMake variable OIIO_idiff_BINARY. One possibility is that, in pxr_register_test, we get the directory containing idiff from this variable, then pass that in to the test runner via the `--post-path` argument. That will tack on the directory to the end of the PATH whenver the test is run. There might be other ways to inject the necessary info into the test runner, but this seems clean enough?